eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -15,14 +19,17 @@ function Get-SwaggerInfo()
         $content = Get-Content .\$AUTOREST_CONFIG_FILE -Raw
         if ($content -match $swaggerInfoRegex)
         {
+            Pop-Location
```

Comment:
This is a very problematic pattern, as you've already seen. Just use `Get-Content "$dir/$AUTOREST_CONFIG_FILE" -raw` above and don't worry about having to push and pop. If you forget to pop in some exit state, bad things can happen. So just avoid it. The only time this really makes sense is when you have to run some utility that needs the current working directory set. That's not the case here. Just use full paths. You're already constructing paths above so do it here as well, and you can do it more simply as I've shown above. `Join-Path` is often overkill.